src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -0,0 +1,420 @@
+ * {@jvms 4.7.31}) and some access flags have no corresponding
+ * modifier, such as {@linkplain #SYNTHETIC synthetic}.
+ *
+ * <p>The values for the constants representing the access and module
+ * flags are taken from sections of <cite>The Java Virtual Machine
+ * Specification</cite> including {@jvms 4.1} (class access and
+ * property modifiers), {@jvms 4.5} (field access and property flags),
+ * {@jvms 4.6} (method access and property flags), {@jvms 4.7.6}
+ * (nested class access and property flags), {@jvms 4.7.24} (method
+ * parameters), and {@jvms 4.7.25} (module flags and requires,
+ * exports, and opens flags).
+ *
+ * <p>The {@linkplain #mask() mask} values for the different access
+ * flags are <em>not</em> distinct. Flags are defined for different
+ * kinds of JVM structures and the same bit position has different
+ * meanings in different contexts. For example, {@code 0x0000_0040}
+ * indicates a {@link #VOLATILE volatile} field but a {@linkplain
+ * #BRIDGE bridge method}; {@code 0x0000_0080} indicates a {@link
+ * #TRANSIENT transient} field but a {@linkplain #VARARGS variable
+ * arity (varargs)} method.
+ *
+ * <p>The access flag constants are ordered by non-decreasing mask
+ * value; that is the mask value of a constant is greater than or
+ * equal to the mask value of an immediate neighbor to its (syntactic)
+ * left. If new constants are added, this property will be
+ * maintained. That implies new constants will not necessarily be
+ * added at the end of the existing list.
+ *
+ * @see java.lang.reflect.Modifier
+ * @see java.lang.module.ModuleDescriptor.Modifier
+ * @see java.lang.module.ModuleDescriptor.Requires.Modifier
+ * @see java.lang.module.ModuleDescriptor.Exports.Modifier
+ * @see java.lang.module.ModuleDescriptor.Opens.Modifier
+ * @see java.compiler/javax.lang.model.element.Modifier
+ * @since 20
+ */
+@SuppressWarnings("doclint:reference") // cross-module link
+public enum AccessFlag {
+    /**
+     * The access flag {@code ACC_PUBLIC}, corresponding to the source
+     * modifier {@link Modifier#PUBLIC public} with a mask value of
+     * <code>{@value "0x%04x" Modifier#PUBLIC}</code>.
+     */
+    PUBLIC(Modifier.PUBLIC, true,
+           Set.of(Location.CLASS, Location.FIELD, Location.METHOD,
+                  Location.INNER_CLASS)),
+
+    /**
+     * The access flag {@code ACC_PRIVATE}, corresponding to the
+     * source modifier {@link Modifier#PRIVATE private} with a mask
+     * value of <code>{@value "0x%04x" Modifier#PRIVATE}</code>.
+     */
+    PRIVATE(Modifier.PRIVATE, true,
+            Set.of(Location.FIELD, Location.METHOD, Location.INNER_CLASS)),
+
+    /**
+     * The access flag {@code ACC_PROTECTED}, corresponding to the
+     * source modifier {@link Modifier#PROTECTED protected} with a mask
+     * value of <code>{@value "0x%04x" Modifier#PROTECTED}</code>.
+     */
+    PROTECTED(Modifier.PROTECTED, true,
+              Set.of(Location.FIELD, Location.METHOD, Location.INNER_CLASS)),
```

Comment:
In both space and startup time, creating separate sets for the same set of Locations is inefficient.